Diagnostics/start-windows-build.ps1

```diff
@@ -0,0 +1,405 @@
+        #dir diagnostics.exe
+    } finally {
+        Pop-Location
+    }
+    Write-Output "Diagnostics was successfully built"
+}
+
+function Generate-ServerListFile {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$FilePath
+    )
+    Write-Output "ServerListFile = $FilePath"
+    $services = Get-Variable |  where {$_.Name.Contains("_SERVICE_NAME")} 
+    foreach ($service in $services) {
+        $service.Value | Out-File $FilePath -Append
+    }
+}
+
+function Generate-DiagnosticsZipFile {
+    Write-Output "Started building Diagnostics binaries"
+    write-output "DIAGNOSTICS_GIT_REPO_DIR: $DIAGNOSTICS_GIT_REPO_DIR" 
+    New-Directory $DIAGNOSTICS_BUILD_BINARIES_DIR
+    Generate-ServerListFile -FilePath $DIAGNOSTICS_SERVICE_LIST_FILE_DIR
+
+    Copy-Item -Path "$DIAGNOSTICS_DCOS_WINDOWS_GIT_REPO_DIR\scripts\detect_ip.ps1" -Destination $DIAGNOSTICS_BUILD_BINARIES_DIR
+    Copy-Item -Path "$DIAGNOSTICS_MESOS_JENKINS_GIT_REPO_DIR\diagnostics\config" -Destination $DIAGNOSTICS_BUILD_BINARIES_DIR -Recurse
+
+    Copy-Item -Force -Path "$DIAGNOSTICS_GIT_REPO_DIR\*.exe" -Destination "$DIAGNOSTICS_BUILD_BINARIES_DIR\"
+    Compress-Files -FilesDirectory "$DIAGNOSTICS_BUILD_BINARIES_DIR\" -Filter "*.*" -Archive "$DIAGNOSTICS_BUILD_BINARIES_DIR\diagnostics.zip"
+    Write-Output "Diagnostic zip file: $DIAGNOSTICS_BUILD_BINARIES_DIR\diagnostics.zip"
+    Write-Output "Diagnostics files were successfully generated"
+}
+function Copy-FilesToRemoteServer {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$LocalFilesPath,
+        [Parameter(Mandatory=$true)]
+        [string]$RemoteFilesPath
+    )
+    Write-Output "Started copying files from $LocalFilesPath to remote location at ${server}:${RemoteFilesPath}"
+    Start-SCPCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY `
+                     -LocalPath $LocalFilesPath -RemotePath $RemoteFilesPath
+}
+
+function New-RemoteDirectory {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$RemoteDirectoryPath
+    )
+    $remoteCMD = "if [[ -d $RemoteDirectoryPath ]]; then rm -rf $RemoteDirectoryPath; fi; mkdir -p $RemoteDirectoryPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+
+function New-RemoteSymlink {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$RemotePath,
+        [Parameter(Mandatory=$false)]
+        [string]$RemoteSymlinkPath
+    )
+    $remoteCMD = "if [[ -h $RemoteSymlinkPath ]]; then unlink $RemoteSymlinkPath; fi; ln -s $RemotePath $RemoteSymlinkPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+function Get-DiagnosticsBuildRelativePath {
+    if($ReviewID) {
+        return "review/$ReviewID"
+    }
+    $repositoryOwner = $GitURL.Split("/")[-2]
+    $mesosCommitID = Get-LatestCommitID
```

Comment:
This is exactly the pattern that will be used when uploading diagnostics builds to our CI server. Please just rename the variable `$mesosCommitID` to `$diagnosticsCommitID`